src/common/hook/useQueue/utils/publish.tsx

```diff
@@ -124,25 +150,17 @@ export const groupDocumentsIntoJobs = async (
   const jobs: PublishingJob[] = [];
   // Process all verifiable documents with document store first
   for (const contractAddress of documentStoreAddresses) {
-    const firstRawDocument = verifiableDocumentsWithDocumentStore[contractAddress][0];
-    // eslint-disable-next-line @typescript-eslint/no-shadow
-    const rawDocuments = verifiableDocumentsWithDocumentStore[contractAddress].map((doc) => doc.rawDocument);
-    const wrappedDocuments = await wrapDocuments(rawDocuments);
-    const firstWrappedDocument = wrappedDocuments[0];
-    const merkleRoot = utils.getMerkleRoot(firstWrappedDocument);
-
-    jobs.push({
-      type: firstRawDocument.type,
-      nonce,
-      contractAddress,
-      documents: verifiableDocumentsWithDocumentStore[contractAddress].map((doc, index) => ({
-        ...doc,
-        wrappedDocument: wrappedDocuments[index],
-      })),
-      merkleRoot: merkleRoot,
-      payload: {},
+    const verifiableDocumentsV2 = verifiableDocumentsWithDocumentStore[contractAddress].filter((docs) => {
+      return utils.isRawV2Document(docs.rawDocument);
     });
-    nonce += TX_NEEDED_FOR_VERIFIABLE_DOCUMENTS;
+    const verifiableDocumentsV3 = verifiableDocumentsWithDocumentStore[contractAddress].filter((docs) => {
+      return utils.isRawV3Document(docs.rawDocument);
+    });
+
+    await processVerifiableDocument(jobs, nonce, contractAddress, verifiableDocumentsV2);
+    if (verifiableDocumentsV2.length !== 0) nonce += TX_NEEDED_FOR_VERIFIABLE_DOCUMENTS;
+    await processVerifiableDocument(jobs, nonce, contractAddress, verifiableDocumentsV3);
+    if (verifiableDocumentsV3.length !== 0) nonce += TX_NEEDED_FOR_VERIFIABLE_DOCUMENTS;
```

Comment:
wont this cause the nonce to be repeated for the different types of documents?

Reply:
the nonce are incremented per transaction so it won't clash

Reply:
sorry for some reason this PR has been forced close, I created other new PR for this branch merging